File: Arrays/Algorithms/Searching/Sorted/binary_iterative_search.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(const vector<int>& arr, int x) {
	int l = 0, r = arr.size() - 1;
	while (l <= r) {
		int m = l + (r - l) / 2;
		if (arr[m] == x) return m;
		if (arr[m] < x) l = m + 1;
		else r = m - 1;
	}
	return -1;
}

int main() {
	int n, x;
	cin >> n >> x;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	cout << binarySearch(arr, x) << endl;
	return 0;
}

File: Arrays/Algorithms/Searching/Sorted/binary_iterative_search.py
def binary_search(arr, x):
	l, r = 0, len(arr) - 1
	while l <= r:
		m = l + (r - l) // 2
		if arr[m] == x:
			return m
		elif arr[m] < x:
			l = m + 1
		else:
			r = m - 1
	return -1

if __name__ == "__main__":
	n, x = map(int, input().split())
	arr = list(map(int, input().split()))
	print(binary_search(arr, x))
